backend/satellite_tools/repo_plugins/deb_src.py

```diff
@@ -94,6 +94,66 @@ def __init__(
         self.url = url
         parts = url.rsplit('/dists/', 1)
         self.base_url = [parts[0]]
+
+        parsed_url = urlparse.urlparse(url)
+        query = urlparse.parse_qsl(parsed_url.query)
+        new_query = []
+        suite = None
+        component = None
+        arch = None
+        for qi in query:
+            if qi[0] == "uyuni_suite":
+                suite = qi[1]
+            elif qi[0] == "uyuni_component":
+                component = qi[1]
+            elif qi[0] == "uyuni_arch":
+                arch = qi[1]
+            else:
+                new_query.append(qi)
+        if suite:
+            parsed_url = parsed_url._replace(query=urlparse.urlencode(new_query))
+            base_url = urlparse.urlunparse(parsed_url)
+            path_list = parsed_url.path.split("/")
+            log2(0, 0, "Base URL: {}".format(base_url))
+            log2(0, 0, "Suite: {}".format(suite))
+            log2(0, 0, "Component: {}".format(component))
+            if "/" not in suite:
+                path_list.append("dists")
+            path_list.extend(suite.split("/"))
+            if component:
+                path_list.extend(component.split("/"))
+            if "/" not in suite:
+                if arch is None:
+                    rhnSQL.initDB()
+                    h = rhnSQL.prepare("""
+                                       SELECT ca.label AS arch_label
+                                       FROM rhnChannel AS c
+                                       LEFT JOIN rhnChannelArch AS ca
+                                           ON c.channel_arch_id = ca.id
+                                       WHERE c.label = :channel_label
+                                       """)
```

Comment:
BTW, this string contains the (common) leading white space. There is `textwrap.dedent()` if that's not wanted .Here I think it's not a big deal, just caught my attention now.

```python
>>> if True:
...     if True:
...         if True:
...             print("""
...                   SELECT something
...                   FROM somewhere
...                   WHERE condition
...                   """)
... 

                  SELECT something
                  FROM somewhere
                  WHERE condition
>>> import textwrap
>>> if True:
...     if True:
...         if True:
...             print(textwrap.dedent("""
...                                   SELECT something
...                                   FROM somewhere
...                                   WHERE condition
...                                   """))
... 

SELECT something
FROM somewhere
WHERE condition


```